main.go

```diff
@@ -95,13 +136,41 @@ func (g Gaz) Run() {
 
 		// register /version to return the build version
 		g.Router.HandleFunc("/version", VersionHTML).Methods("GET")
-
-		port := viper.GetInt("http.port")
-		Sugar.Infof("Starting http server on port %d", port)
-		err := http.ListenAndServe(fmt.Sprintf(":%d", port), g.Router)
+		Sugar.Infof("Starting HTTP server on :%d", g.httpPort)
+		err := http.Serve(*g.httpListener, g.Router)
 		if err != nil {
-			Sugar.Errorf("Cannot start HTTP server on :%d, %+v", port, err)
+			Sugar.Errorf("Cannot updater HTTP server on :%d, %+v", g.httpPort, err)
 			panic(err)
 		}
 	}()
 }
+
+func (g Gaz) Register(d *ServiceDefinition) (string, error) {
```

Comment:
Shouldn't it be on (g *Gaz) instead of (g Gaz)?

Reply:
it seems that once built, the Gaz object is not modified, so should we remove the pointer from all methods?